yt_dlp/extractor/ant1newsgr.py

```diff
@@ -0,0 +1,184 @@
+            'id': '_xvg/m_cmbatw=',
+            'ext': 'mp4',
+            'title': 'md5:a93e8ecf2e4073bfdffcb38f59945411',
+            'timestamp': 1603092840,
+            'upload_date': '20201019',
+            'thumbnail': 'https://ant1media.azureedge.net/imgHandler/1920/756206d2-d640-40e2-b201-3555abdfc0db.jpg',
+        },
+    }, {
+        'url': 'https://ant1news.gr/Society/article/620286/symmoria-anilikon-dikigoros-thymaton-ithelan-na-toys-apoteleiosoyn',
+        'info_dict': {
+            'id': '620286',
+            'title': 'md5:91fe569e952e4d146485740ae927662b',
+        },
+        'playlist_mincount': 2,
+        'params': {
+            'skip_download': True,
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        info = self._search_json_ld(webpage, video_id, expected_type='NewsArticle')
+        embed_urls = list(Ant1NewsGrEmbedIE._extract_urls(webpage, url, **info))
+        if not embed_urls:
+            raise ExtractorError('no videos found for %s' % video_id)
+        if len(embed_urls) == 1:
+            return self.url_result(embed_urls[0], ie=Ant1NewsGrEmbedIE.ie_key(),
+                                   video_title=info['title'])
+        return self.playlist_from_matches(
+            embed_urls, video_id, info['title'], ie=Ant1NewsGrEmbedIE.ie_key())
+
+
+class Ant1NewsGrEmbedIE(Ant1NewsGrBaseIE):
+    IE_NAME = 'ant1newsgr:embed'
+    IE_DESC = 'ant1news.gr embedded videos'
+    _VALID_URL = r'''(?x)https?://(?:[a-zA-Z0-9\-]+\.)?
+        (?:antenna|ant1news)\.gr/templates/pages/player
+        \?(?:(?:cid=(?P<id>[^&#]+)|[^&=#]+=[^&#]+)&?)+'''
+    _API_PATH = '/news/templates/data/jsonPlayer'
+
+    _TEST = {
+        'url': 'https://www.antenna.gr/templates/pages/player?cid=3f_li_c_az_jw_y_u=&w=670&h=377',
+        'md5': '12872b12af18b5dbf76528786728de8c',
+        'info_dict': {
+            'id': '3f_li_c_az_jw_y_u=',
+            'ext': 'mp4',
+            'title': 'md5:a30c93332455f53e1e84ae0724f0adf7',
+        },
+    }
+
+    @classmethod
+    def _extract_urls(cls, webpage, origin_url=None, **parent_info):
+        # make the scheme in _VALID_URL optional
+        _URL_RE = r'(?:https?:)?//' + cls._VALID_URL.split('://', 1)[1]
+        # simplify the query string part of _VALID_URL; after extracting iframe
+        # src, the URL will be matched again
+        _URL_RE = _URL_RE.split(r'\?', 1)[0] + r'\?(?:(?!(?P=_q1)).)+'
+        EMBED_RE = r'''(?x)
+            <iframe[^>]+?src=(?P<_q1>%(quot_re)s)(?P<url>%(url_re)s)(?P=_q1)
+        ''' % {'quot_re': r'["\']', 'url_re': _URL_RE}
+        for mobj in re.finditer(EMBED_RE, webpage):
+            url = unescapeHTML(mobj.group('url'))
+            if url.startswith('//'):
+                scheme = urlparse(origin_url).scheme if origin_url else 'https'
+                url = '%s:%s' % (scheme, url)
+            if not cls.suitable(url):
```

Comment:
maybe just do `self._proto_relative_url` in generic to avoid this complexity

Reply:
`_extract_urls` is a `classmethod`, so I am afraid it won't work after all (due to the `prefer_insecure` parameter lookup).

Reply:
I completely removed this block since
* `_proto_relative_url` is subsequently called in `playlist_from_matches`
* it is now also called in `Ant1NewsGrArticleIE._real_extract`